### PR TITLE
feat(share/shrex): collect errors from shrex getter

### PR DESCRIPTION
**Pull request by walldiss**
## Overview

Shrex getter client needs to collect and propagate request errors to the caller. Also introduce per error logging.

### CODE REVIEW

File: share/getters/shrex.go [hunk 1]
```diff
@@ -62,32 +64,45 @@ func (sg *ShrexGetter) GetShare(ctx context.Context, root *share.Root, row, col
...
 			return eds, nil
 		case context.DeadlineExceeded:
-			log.Debugw("request exceeded deadline, trying with new peer", "datahash", root.String())
 		case p2p.ErrInvalidResponse:
 			setStatus(peers.ResultBlacklistPeer)
 		default:
 			setStatus(peers.ResultCooldownPeer)
 		}
+
+		err = multierr.Append(err, getErr)
```
Comment (Wondertan) [inline share/getters/shrex.go:99]: Join 🤤

**Review by Wondertan — APPROVED** [review]
Still disagree that we should return an error for each attempt, especially if the errors are the same on each try, but it does not hurt so approve

File: share/getters/shrex.go (hunk 1 above)
Comment (walldiss) [reply share/getters/shrex.go:99]: Found 1.20 enthusiast.

BTW good point about collecting duplicate errors. I"ll add extra filter

**Review by Wondertan — DISMISSED** [review]
Very nice

**Review by vgonkivs — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

File: share/getters/shrex.go [hunk 2]
```diff
@@ -97,33 +113,46 @@ func (sg *ShrexGetter) GetSharesByNamespace(
...
-			log.Debugw("request exceeded deadline, trying with new peer", "datahash", root.String())
 		case p2p.ErrInvalidResponse:
 			setStatus(peers.ResultBlacklistPeer)
 		default:
 			setStatus(peers.ResultCooldownPeer)
 		}
+
+		if !ErrorContains(err, getErr) {
+			err = multierr.Append(err, getErr)
+		}
```
Comment (renaynay) [inline share/getters/shrex.go:150]: I'm confused -- so if we get two peers who return to us an `ErrInvalidResponse` for example, we would only collect that err once?

**Review by renaynay — APPROVED** [review]

**Pull request merged**